controllers/argocd/keycloak.go

```diff
@@ -1248,9 +1362,24 @@ func (r *ReconcileArgoCD) reconcileKeycloakForOpenShift(cr *argoprojv1a1.ArgoCD)
 			log.Info(fmt.Sprintf("Successfully created keycloak realm for ArgoCD %s in namespace %s",
 				cr.Name, cr.Namespace))
 
+			// OAuthClient configuration does not get deleted from previous instances occasionally.
+			// It is safe to delete before updating the OIDC config.
+			err = deleteOAuthClient(cr)
```

Comment:
This is worrisome. Do we know why the previous instance's oauthclient is not deleted?

Reply:
I noticed this issue once a very long time back but I thought it is issue with my configuration, I noticed this again recently. I did not find any issue on OpenShift API. 

So to be 100% sure, I added this double check.